dotnet-manual-instrumentation/Azure.Functions/ManuallyInstrumentedHttpTrigger.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web.Http;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using OpenTracing.Manually.Instrumented.Library;
+using OpenTracing.Util;
+using SignalFx.Tracing;
+
+namespace Azure.Functions
+{
+    public static class ManuallyInstrumentedHttpTrigger
+    {
+        private static readonly string SamplerServerUrl = $"http://localhost:19999{SampleServer.RequestPath}/";
+        
+        // All configurations for the SignalFxTracer are being captured from environment
+        // variables defined in local.settings.json. This static is just used to trigger
+        // its initialization, from this point all OpenTracing.Util.GlobalTracer.Instance
+        // will be using the SignalFxTracer.
+        private static readonly Tracer SignalFxTracer = Tracer.Instance;
+        
+        static ManuallyInstrumentedHttpTrigger()
+        {
+            // Simulate the starting of a server to server background requests. 
+            var server = new SampleServer();
+            server.Start(SamplerServerUrl);
+        }
+        
+        [FunctionName("ManuallyInstrumentedHttpTrigger")]
+        public static async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "get", "post", Route = null)] HttpRequest req,
+            ILogger log)
+        {
+            var tracer = GlobalTracer.Instance;
+            using var scope = tracer.BuildSpan("HttpTrigger").StartActive();

```

Comment:
Would it make sense to try to extract any propagated context here like you do in the manually instrumented library?

Reply:
Good idea. In principle, besides the typical propagation like B3, there is RequestId (a GUID) in Azure.Functions that perhaps we could use here. Since we already have an example extracting B3 I will put in the backlog the possibility of using the RequestId.